source/opt/copy_prop_arrays.cpp

```diff
@@ -806,6 +814,11 @@ uint32_t CopyPropagateArrays::MemoryObject::GetNumberOfMembers() {
     const analysis::Constant* length_const =
         context->get_constant_mgr()->FindDeclaredConstant(
             array_type->LengthId());
+
+    if (length_const == nullptr) {
+      // This can happen if the length is an OpSpecConstant.
+      return 0;
+    }
```

Comment:
I cannot create a test where this makes a difference because it is only called for OpCompositeConstruct, which is not possible if the array length is unknown.

Reply:
However, I want to function to be useful more generally, so I still want to output the correct value.

Reply:
I fixed this by refactoring the code so that it is that both Insert and Construct use the same code to get the number of elements.